identity-account/src/updates/update.rs

```diff
@@ -0,0 +1,465 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crypto::signatures::ed25519;
+
+use identity_core::common::Fragment;
+use identity_core::common::Object;
+use identity_core::common::Timestamp;
+use identity_core::crypto::KeyPair;
+use identity_core::crypto::KeyType;
+use identity_core::crypto::PublicKey;
+use identity_did::did::CoreDIDUrl;
+use identity_did::did::DID;
+use identity_did::service::Service;
+use identity_did::service::ServiceEndpoint;
+use identity_did::verification::MethodRef;
+use identity_did::verification::MethodRelationship;
+use identity_did::verification::MethodScope;
+use identity_did::verification::MethodType;
+use identity_iota::did::IotaDID;
+use identity_iota::did::IotaDIDUrl;
+use identity_iota::did::IotaDocument;
+use identity_iota::did::IotaVerificationMethod;
+
+use crate::account::Account;
+use crate::error::Result;
+use crate::identity::DIDLease;
+use crate::identity::IdentitySetup;
+use crate::identity::IdentityState;
+use crate::storage::Storage;
+use crate::types::Generation;
+use crate::types::KeyLocation;
+use crate::types::MethodSecret;
+use crate::updates::UpdateError;
+
+// Method types allowed to sign a DID document update.
+pub const UPDATE_METHOD_TYPES: &[MethodType] = &[MethodType::Ed25519VerificationKey2018];
+pub const DEFAULT_UPDATE_METHOD_PREFIX: &str = "sign-";
+
+fn key_to_method(type_: KeyType) -> MethodType {
+  match type_ {
+    KeyType::Ed25519 => MethodType::Ed25519VerificationKey2018,
+  }
+}
```

Comment:
I think this is only used in one place now, maybe remove the function and inline the code?